packages/web/src/pages/dashboard-page/components/ArtistContentSection.tsx

```diff
@@ -0,0 +1,461 @@
+        icon: IconVisibilityPublic,
+        value: AlbumFilters.PUBLIC
+      }
+    ]
+    if (premiumAlbums.length) {
+      filterButtonAlbumOptions.push({
+        id: AlbumFilters.PREMIUM,
+        label: messages.premium,
+        icon: IconCart,
+        value: AlbumFilters.PREMIUM
+      })
+    }
+    if (hiddenAlbums.length) {
+      filterButtonAlbumOptions.push({
+        id: AlbumFilters.HIDDEN,
+        label: messages.hidden,
+        icon: IconVisibilityHidden,
+        value: AlbumFilters.HIDDEN
+      })
+    }
+
+    return isTracks ? filterButtonTrackOptions : filterButtonAlbumOptions
+  }, [
+    isTracks,
+    collectibleGatedTracks,
+    hiddenAlbums,
+    hiddenTracks,
+    premiumAlbums,
+    premiumTracks,
+    specialAccessTracks
+  ])
+
+  if (filterText) {
+    if (isTracks) {
+      filteredTracks = filteredTracks.filter((data) =>
+        data.name.toLowerCase().includes(filterText.toLowerCase())
+      )
+    } else {
+      filteredAlbums = filteredAlbums.filter((data) =>
+        data.name.toLowerCase().includes(filterText.toLowerCase())
+      )
+    }
+  }
+
+  if (!tracks.length && !albums.length) return null
+
+  return (
+    <Paper w='100%' direction='column' mt='xl'>
+      <Flex ph='2xl' pv='l' justifyContent='space-between'>
+        <Flex gap='2xl'>
+          {shouldShowPills ? (
+            <Flex gap='s'>
+              <SelectablePill
+                isSelected={selectedPill === Pills.TRACKS}
+                label={messages.tracks}
+                size='large'
+                onClick={() => onClickPill(Pills.TRACKS)}
+              />
+              <SelectablePill
+                isSelected={selectedPill === Pills.ALBUMS}
+                label={messages.albums}
+                size='large'
+                onClick={() => onClickPill(Pills.ALBUMS)}
+              />
+            </Flex>
+          ) : null}
+          {
+            // Only show filter button if there are multiple sections for the selected content type
+            (isTracks && !hasOnlyOneTrackSection) ||
+            (!isTracks && !hasOnlyOneAlbumSection) ? (
+              <FilterButton
+                onSelect={handleSelectFilter}
+                selection={isTracks ? selectedTrackFilter : selectedAlbumFilter}
+                label={messages.allReleases}
+                options={filterButtonOptions}
+                popupAnchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                popupTransformOrigin={{ vertical: 'top', horizontal: 'left' }}
+              />
+            ) : null
+          }
+        </Flex>
+        <Flex>
+          <TextInput
+            placeholder={messages.search(selectedPill)}
+            label={messages.search(selectedPill)}
+            value={filterText}
+            onChange={handleFilterChange}
+            size={TextInputSize.SMALL}
+            startIcon={IconSearch}
+          />
+        </Flex>
+      </Flex>
+      {selectedPill === Pills.TRACKS ? (
```

Comment:
```suggestion
      {isTracks ? (
```

Reply:
If these are truly mutually exclusive, I wonder if we could push some of the logic down into two separate components? Either we could
- Duplicate the entire file and make one for tracks and one for albums
- Manage the filter states and general table layout in the parent, but push the data filtering down to an albums and a tracks table wrapper

Reply:
Yep agreed. Seems like the only things that live at this level are the currently selected pill and the filter text. Everything else could go into nested components which maintain a list of tracks or albums and then render a table with that state.

Reply:
great feedback, i missed forest for trees, updated.